server/api.go

```diff
@@ -272,3 +273,150 @@ func (d dexAPI) RevokeRefresh(ctx context.Context, req *api.RevokeRefreshReq) (*
 
 	return &api.RevokeRefreshResp{}, nil
 }
+
+// parseConnConfig will parse the provided connector config to ensure it is correct.
+func parseConnConfig(connType string, config []byte) error {
+	f, ok := ConnectorsConfig[connType]
+	if !ok {
+		return fmt.Errorf("unknown connector type %q", connType)
+	}
+
+	connConfig := f()
+	if len(config) != 0 {
+		if err := json.Unmarshal(config, connConfig); err != nil {
```

Comment:
OpenConnector() is already called in the getConnector() (https://github.com/coreos/dex/blob/master/server/server.go#L448) which is called in ther server/handlers.go.  Wouldn't it be redundant to call it again here?